.vscode/InlineExpectationsTests.code-snippets

```diff
@@ -0,0 +1,60 @@
+{
```

Comment:
We haven't added any `.code-snippets` in the codeql repo yet, and I'm not aware of a policy regarding this. While I'm personally a big fan of this particular snippet, I don't know if the inline expectation test framework has enough of a userbase to justify putting it in the repo. Do you have any opinion about this?

Reply:
I expected this discussion to happen - my main counterpoint is that these snippets are restricted to cpp files in our repo, which is only the tests, so most of our users will never see the snippets.
Those who do may or might not work on dataflow/taint tracking tests and then they may or may not find them useful.

I mainly wanted to share these snippets internally, as I believe they will be useful for everybody on the team, given that we evolve dataflow/taint tracking regularly.

Reply:
If one person other than @criemen finds the snippets useful, then I won't object to checking them in. I'm not that person: I think the burden of typing these annotations out will be far smaller when most of them have been converted and we only need to add annotations for new cases or sometimes edit existing ones.